### PR TITLE
feat(build-config): use shorter classnames in production

### PR DESCRIPTION
**Pull request by HerrBertling**
fixes https://github.com/xing/hops/issues/325

This pull request closes issue #325

## Current state

The generated classnames are the same for all instances. This leads to quite long classnames in production.

## Changes introduced here

Shorter classnames in production. Yay 🎉 

## Checklist

* [x] All commit messages adhere to the [appropriate format](https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit) in order to trigger a correct release
* [x] All code is written in plain ECMAScript v5 and is formatted using [prettier](https://prettier.io)
* [ ] Necessary unit tests are added in order to ensure correct behavior
* [ ] Documentation has been added

### CODE REVIEW

**Comment by dmbch** [comment]
Thanks for your PR! 🚀 

As discussed IRL, we will have to find a way to also implement this for server rendering. Maybe we really ought to simply rely on `NODE_ENV` and 'minify' the class name if it is set to `production`..?

File: packages/build-config/sections/module-rules/postcss.js [hunk 1]
```diff
@@ -30,7 +30,9 @@ exports.build = {
     use: [
       {
         loader: require.resolve('css-loader'),
-        options: cssLoaderOptions,
+        options: Object.assign({}, cssLoaderOptions, {
+          localIdentName: '[hash:base64:8]',
```
Comment (dmbch) [inline packages/build-config/sections/module-rules/postcss.js]: We will also have to do this for node to make sure that the generated markup is equal no matter where an in what env it is rendered.

**Comment by systemboogie** [comment]
I like that 👍 
However, I want to point out that we have to properly communicate this (actually "non-breaking") change before merging because some end-to-end tests may rely on classnames.

**Comment by HerrBertling** [comment]
I changed it to use the `NODE_ENV` to decide how to handle the classes :)

**Comment by dmbch** [comment]
@systemboogie good point. How should we proceed? Ask @matthias-reis to include in in our next newsletter? Add a note to the README? 🤔

**Comment by systemboogie** [comment]
@dmbch 
There is a proposal in Malt (1773) that has already been worked on. I would say we first have to get that through the door. After that, we can spread the news via developer/qa lists, communities of practice and in person. Then we should give the teams some time to use the new testId implementation and adjust their tests.

**Comment by HerrBertling** [comment]
So we'll keep this open? Or merge it and advise people to update only if…?

**Comment by systemboogie** [comment]
Maybe it is a good idea to make it an "opt-in" feature? That way we could merge and teams could decide whether to use it or not. Otherwise I would not merge at the moment.

**Comment by HerrBertling** [comment]
Sounds good :)

**Comment by HerrBertling** [comment]
I'll close here and redo this as a feature :)

**Comment by HerrBertling** [comment]
Or not, as I talked with @ZauberNerd and @robin-drexler who'd rather keep the footprint of this low instead of adding a special opt-in feature for this setting :)

**Comment by dmbch** [comment]
Repetitive, long class names counterintuitively lead to smaller gzipped downloads, therefore I am closing this for the time being.

**Pull request closed without merging**